### ISSUE TITLE
Repeated notification when location disabled due to redundant checks

### ISSUE THREAD

**Issue by tremblerz** [Path-Check/covid-safe-paths#656]
## Version

- Type of phone: Android / iPhone
- OS Version:
- Model: iPhone 11 (simulator)

## Steps to reproduce

Go to Settings, turn off location
1. You get notification
2. now you pull the notification drawer and swipe on notification to get rid of it
3. Drawer closes and the app comes in foreground and another notification gets triggered
4. Go to 2.

## Expected Behavior

Any event should generate notification only once in general, the way we trigger notification in our app is asynchronously checking states at every screen mount/refresh, we need to shift the logic perhaps

**Comment by eyganeshv** [comment]
I will check this issue. We need detail clarification on this issue.

**Comment by summetj** [comment]
@eyganeshv , post questions you have about this issue here and we will do our best to address them.

**Comment by NaziyaNazir** [comment]
This issue is seen for Android also.

**Comment by ArchanaNair** [comment]
@NaziyaNazir, reproducible steps same for android?

**Comment by NaziyaNazir** [comment]
@ArchanaNair Yes, the repro steps are same.

**Comment by ArchanaNair** [comment]
May I know the OS version? I am unable to reproduce it on Android, verifying it on OS version 9

**Comment by NaziyaNazir** [comment]
@ArchanaNair  OS version is Android 9.

**Comment by ArchanaNair** [comment]
@NaziyaNazir , I could see the issue when app goes to background and again comes to foreground, but not in the scenario which is given in reproducible step.

**Comment by summetj** [comment]
This is starting to sound like a low priority issue, anybody disagree?
 And is it strictly a UI/UX issue, or an issue with how the location subsystem works?

**Comment by ArchanaNair** [comment]
> @NaziyaNazir , I could see the issue when app goes to background and again comes to foreground, but not in the scenario which is given in reproducible step

@tremblerz @NaziyaNazir  included a fix for this in android in PR #764

**Comment by tstirrat** [comment]
I could not get this to reproduce on iOS or Android anymore. I believe it was fixed in #749

**Comment by ArchanaNair** [comment]
Yes the issue is fixed in android also.
